service/dps/storm/topologies/indexing/src/main/java/eu/europeana/cloud/service/dps/storm/topologies/indexing/IndexingTopology.java

```diff
@@ -0,0 +1,174 @@
+    private final String FILE_STREAM = InputDataType.FILE_URLS.name();
+    public static final String SUCCESS_MESSAGE = "Record is indexed correctly";
+
+    public IndexingTopology(String defaultPropertyFile, String providedPropertyFile, String defaultIndexingPropertiesFile, String providedIndexingPropertiesFile) {
+        PropertyFileLoader.loadPropertyFile(defaultPropertyFile, providedPropertyFile, topologyProperties);
+        PropertyFileLoader.loadPropertyFile(defaultIndexingPropertiesFile, providedIndexingPropertiesFile, indexingProperties);
+        brokerHosts = new ZkHosts(topologyProperties.getProperty(TopologyPropertyKeys.INPUT_ZOOKEEPER_ADDRESS));
+    }
+
+    public final StormTopology buildTopology(String indexingTopic, String ecloudMcsAddress) {
+        Map<String, String> routingRules = new HashMap<>();
+        routingRules.put(DATASET_STREAM, DATASET_STREAM);
+        routingRules.put(FILE_STREAM, FILE_STREAM);
+        ReadFileBolt retrieveFileBolt = new ReadFileBolt(ecloudMcsAddress);
+
+        SpoutConfig kafkaConfig = new SpoutConfig(brokerHosts, indexingTopic, "", "storm");
+        kafkaConfig.scheme = new SchemeAsMultiScheme(new StringScheme());
+        kafkaConfig.ignoreZkOffsets = true;
+        kafkaConfig.startOffsetTime = kafka.api.OffsetRequest.LatestTime();
+        TopologyBuilder builder = new TopologyBuilder();
+        KafkaSpout kafkaSpout = new CustomKafkaSpout(kafkaConfig, topologyProperties.getProperty(TopologyPropertyKeys.CASSANDRA_HOSTS),
+                Integer.parseInt(topologyProperties.getProperty(TopologyPropertyKeys.CASSANDRA_PORT)),
+                topologyProperties.getProperty(TopologyPropertyKeys.CASSANDRA_KEYSPACE_NAME),
+                topologyProperties.getProperty(TopologyPropertyKeys.CASSANDRA_USERNAME),
+                topologyProperties.getProperty(TopologyPropertyKeys.CASSANDRA_SECRET_TOKEN));
+
+
+        builder.setSpout(TopologyHelper.SPOUT, kafkaSpout,
+                ((int) Integer.parseInt(topologyProperties.getProperty(TopologyPropertyKeys.KAFKA_SPOUT_PARALLEL))))
+                .setNumTasks(
+                        ((int) Integer.parseInt(topologyProperties.getProperty(TopologyPropertyKeys.KAFKA_SPOUT_NUMBER_OF_TASKS))));
+
+        builder.setBolt(TopologyHelper.PARSE_TASK_BOLT, new ParseTaskBolt(routingRules),
+                ((int) Integer
+                        .parseInt(topologyProperties.getProperty(TopologyPropertyKeys.PARSE_TASKS_BOLT_PARALLEL))))
+                .setNumTasks(
+                        ((int) Integer.parseInt(topologyProperties.getProperty(TopologyPropertyKeys.PARSE_TASKS_BOLT_NUMBER_OF_TASKS))))
+                .shuffleGrouping(TopologyHelper.SPOUT);
+
+        builder.setBolt(TopologyHelper.READ_DATASETS_BOLT, new ReadDatasetsBolt(),
+                ((int) Integer
+                        .parseInt(topologyProperties.getProperty(TopologyPropertyKeys.READ_DATASETS_BOLT_PARALLEL))))
+                .setNumTasks(
```

Comment:
1- There is many usage of TopologyPropertyKeys static fields, maybe we can import them statically and spare using the class name again !
2- The same applies to TopologyHelper